internal/database/connect.go

```diff
@@ -1,13 +1,11 @@
 package database
 
 import (
-	"github.com/source-academy/stories-backend/internal/config"
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
 )
 
-func Connect(conf *config.DatabaseConfig) (*gorm.DB, error) {
-	dsn := conf.ToDataSourceName()
+func Connect(dsn string) (*gorm.DB, error) {
```

Comment:
To me, moving the DSN generation outside seems to initially break abstraction, since this DSN is specific to postgres.

Was there a reason this change was done? Was it testability or something else?

Reply:
Right, I was thinking about this as well. Essentially, I need another function that connects to just the db server, since the database with the given name may not exist in the db server.

As you mentioned earlier, I could write another similar function and move all the create functionalities to the script folder?